include/nlohmann/json.hpp

```diff
@@ -3522,15 +3603,26 @@ class basic_json
             // fill up array with null values if given idx is outside range
             if (idx >= m_value.array->size())
             {
-                m_value.array->insert(m_value.array->end(),
-                                      idx - m_value.array->size() + 1,
-                                      basic_json());
+#if JSON_DIAGNOSTICS
+                // remember array size before resizing
+                const auto previous_size = m_value.array->size();
+#endif
+
+                m_value.array->resize(idx + 1);
+
+#if JSON_DIAGNOSTICS
+                // set parent for values added above
+                for (auto i = previous_size; i <= idx; ++i)
+                {
+                    set_parent(m_value.array->operator[](i));
+                }
+#endif
```

Comment:
I think if you wanted this could be done as `set_parents(m_value.array->begin() + previous_size, idx + 1 - previous_size);`

Reply:
I thought about this, too, but thought such code would be too ugly. But since you had the same idea, I think I'll get rid of that loop.

Reply:
Fixed in b0d8628c498d1a2be3391a1949886843746b9878.